--- a/drivers/gpu/drm/amd/pm/swsmu/inc/smu_types.h	2026-04-27 15:50:41.000000000 +0000
+++ b/drivers/gpu/drm/amd/pm/swsmu/inc/smu_types.h	2026-05-22 17:14:33.239398004 +0000
@@ -242,6 +242,26 @@
 	__SMU_DUMMY_MAP(BoardPowerCalibration),   \
 	__SMU_DUMMY_MAP(RequestGfxclk),           \
 	__SMU_DUMMY_MAP(ForceGfxVid),             \
+	__SMU_DUMMY_MAP(GetGfxFrequency),         \
+	__SMU_DUMMY_MAP(GetGfxVid),               \
+	__SMU_DUMMY_MAP(SetCoreEnableMask),       \
+	__SMU_DUMMY_MAP(RequestCorePstate),       \
+	__SMU_DUMMY_MAP(QueryCorePstate),         \
+	__SMU_DUMMY_MAP(QueryGfxclk),             \
+	__SMU_DUMMY_MAP(QueryVddcrSocClock),      \
+	__SMU_DUMMY_MAP(QueryDfPstate),           \
+	__SMU_DUMMY_MAP(SetMinDeepSleepGfxclkFreq), \
+	__SMU_DUMMY_MAP(StartTelemetryReporting), \
+	__SMU_DUMMY_MAP(StopTelemetryReporting),  \
+	__SMU_DUMMY_MAP(ClearTelemetryMax),       \
+	__SMU_DUMMY_MAP(QueryActiveWgp),          \
+	__SMU_DUMMY_MAP(InitiateGcRsmuSoftReset), \
+	__SMU_DUMMY_MAP(GfxCacWeightOperation),   \
+	__SMU_DUMMY_MAP(L3CacWeightOperation),    \
+	__SMU_DUMMY_MAP(PackCoreCacWeight),       \
+	__SMU_DUMMY_MAP(SetDriverTableVMID),      \
+	__SMU_DUMMY_MAP(ForceGfxFreq),            \
+	__SMU_DUMMY_MAP(UnForceGfxFreq),          \
 	__SMU_DUMMY_MAP(Spare0),                  \
 	__SMU_DUMMY_MAP(UnforceGfxVid),           \
 	__SMU_DUMMY_MAP(HeavySBR),			\
